docs/development/code_guide.rst

```diff
@@ -77,18 +77,125 @@ Documentation
+submitted it as a pull request, is to just push your code to GitHub. The Travis
+CI integration runs tests on every uploaded commit.
+
+To run the test suite locally, use ``pytest`` from the repository's root
+directory.  This should detect our ``setup.cfg`` file and thus also verify
+examples in docstrings.
+
+
+A few guidelines to on writing neat, readable and useful tests:
+
 * Unit tests should be provided for all methods when possible.
 
-* Bugs should be turned into test cases.  
+* Solved bugs should be turned into test cases.
   
 * The Travis CI integration on GitHub runs tests whenever pull
   requests to PlasmaPy are updated.  The pytest module may used on a
   local computer.
   
 * Tests are run frequently during code development, and slow tests may
-  interrupt the flow of a contributor.  Tests should be efficient
-  except as needed.
+  interrupt the flow of a contributor.  Tests should be minimal, sufficient enough to
+  be complete and as efficient as possible.
+
+
+Assert statements
+-----------------
+* Pytest runs tests by checking ``assert`` statements, so this is sufficient:
+
+.. code-block:: python
+
+  def test_universe_is_sane():
+      assert 2 + 2 == 4
+
+However, making assertions descriptive is better in most cases:
+
+.. code-block:: python
+
+  def test_universe_is_sane():
+      assert 2 + 2 == 4, "Addition is broken. Reinstall the universe and reboot."
+
+pytest should display the value of the ``2 + 2`` expression, but the value can be added to the thrown string:
+
+.. code-block:: python
+
+  def test_universe_is_sane():
+      assert 2 + 2 == 4, f"Addition is broken, 2 + 2 giving {2 + 2}. Reinstall the universe and reboot."
+
+A note on test independence and parametrization
+-----------------------------------------------
+
+In this section, we'll discuss the issue of parametrization based on an example
+of a `proof <https://en.wikipedia.org/wiki/Riemann\_hypothesis#Excluded\_middle>`_ of Gauss's class number conjecture.
+
+The proof goes along these lines: 
+* If the generalized Riemann hypothesis is true, the conjecture is true.
+* If the former is false, the latter is also true.
+* Therefore, the latter is true.
+
+One way to use pytest for testing is to write continuous assertions:
+
+.. code-block:: python
+
+  def test_proof_by_riemann_hypothesis():
+       # if this step fails, the test stops
+       assert proof_by_riemann(False) 
+       # and you have to run this again
+       assert proof_by_riemann(True) 
+
+To do this the right way, what you technically should do to make the tests independent:
+
+.. code-block:: python
+
+  def test_proof_if_riemann_false():
+       assert proof_by_riemann(False)
+  def test_proof_if_riemann_true():
+       assert proof_by_riemann(True)
+
+but that's a lot of typing so what you actually do is use pytest parametrization:
+
+.. code-block:: python
+
+  @pytest.mark.parametrize("truth", [True, False])
```

Comment:
I am wondering how bad of an idea it would be to change the variable name `"truth"` here to [`"truthiness"`](https://en.wikipedia.org/wiki/Truthiness).